server/src/test/java/org/cloudfoundry/identity/uaa/oauth/token/UserTokenGranterTest.java

```diff
@@ -188,6 +194,15 @@ protected Authentication validateRequest(TokenRequest request) {
 
     @Test
     public void happy_day() {
+        when(clientAuthenticationProvider.authenticate(any(UsernamePasswordAuthenticationToken.class)))
```

Comment:
What's the purpose of this test? It was a little bit confusing before this change , but now with a mock that throws an exception but the name "happy_day" I can't tell if this is a happy path test or not.